### PR TITLE
Add insertUnique_ to PersistUniqueWrite

### PR DESCRIPTION
**Pull request by lbseale**
Before submitting your PR, check that you've:

- [x] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [x] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddock
- [x] Ran `stylish-haskell` on any changed files.
- [x] Adhered to the code style (see the `.editorconfig` file for details)

After submitting your PR:

- [ ] Update the Changelog.md file with a link to your PR
- [ ] Bumped the version number if there isn't an `(unreleased)` on the Changelog
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by lbseale** [comment]
Fixes issue #1457

File: persistent/Database/Persist/Class/PersistUnique.hs [hunk 1]
```diff
@@ -140,6 +140,37 @@ class (PersistUniqueRead backend, PersistStoreWrite backend) =>
...
+    -- > |1    |SPJ   |40   |
+    -- > +-----+------+-----+
+    -- > |2    |Simon |41   |
+    -- > +-----+------+-----+
+    -- > |3    |Linus |48   |
+    -- > +-----+------+-----+
+    --
+    -- Linus's record was inserted to <#dataset-persist-unique-1 dataset-1>, while SPJ wasn't because SPJ already exists in <#dataset-persist-unique-1 dataset-1>.
+    --
+    -- @since 2.14.4.4
```
Comment (parsonsmatt) [inline persistent/Database/Persist/Class/PersistUnique.hs:164]: As an addition to a class with a default method, this is a minor version bump - let's bundle this with 2.15.0.

**Review by parsonsmatt — CHANGES_REQUESTED** [review]
This is great, thank you! Please make a Changelog entry. I'll relesae this with 2.14.5.0

**Review by parsonsmatt — APPROVED** [review]
Thanks!

**Comment by lbseale** [comment]
Thank you! Proud to have my contribution included

**Pull request merged**